dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java

```diff
@@ -1336,7 +1336,7 @@ public Map<String, Object> registerUser(String userName, String userPassword, St
             putMsg(result, Status.REQUEST_PARAMS_NOT_VALID_ERROR, "two passwords are not same");
             return result;
         }
-        User user = createUser(userName, userPassword, email, 1, "", "", Flag.NO.ordinal());
+        User user = createUser(userName, userPassword, email, -1, "", "", Flag.NO.ordinal());
```

Comment:
If we set -1, it will cause some possible NPE, like 
`String tenantCode = tenantMapper.queryById(loginUser.getTenantId()).getTenantCode();`
If you want to modify it, plz make sure to fix those NPE, either :D

Reply:
This risk is already dealt with by DML. By default, a tenant with an id of -1 is added.

https://github.com/apache/dolphinscheduler/blob/399d0d0ce9be29724988dcbd9a7dc5fbfab44de0/dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.0_schema/mysql/dolphinscheduler_dml.sql#L19-L31